packages/tokens/non-fungible/src/extensions/royalties/mod.rs

```diff
@@ -0,0 +1,99 @@
+mod storage;
+use crate::NonFungibleToken;
+
+mod test;
+
+use soroban_sdk::{Address, Env};
+
+/// Royalties Trait for Non-Fungible Token (ERC2981)
+///
+/// The `NonFungibleRoyalties` trait extends the `NonFungibleToken` trait to
+/// provide the capability to set and query royalty information for tokens. This
+/// trait is designed to be used in conjunction with the `NonFungibleToken`
+/// trait.
+///
+/// This implementation follows the ERC2981 standard for royalties, allowing:
+/// - Setting global royalties for the entire collection
+/// - Setting per-token royalties that override the global setting
+/// - Enforcing a maximum royalty percentage to prevent scams
+/// - Making royalties immutable after minting
+///
+/// `storage.rs` file of this module provides the `NonFungibleRoyalties` trait
+/// implementation for the `Base` contract type. For other contract types (eg.
+/// `Enumerable`, `Consecutive`), the overrides of the `NonFungibleRoyalties`
+/// trait methods can be found in their respective `storage.rs` file.
+///
+/// # Notes
+///
+/// `#[contractimpl]` macro requires even the default implementations to be
+/// present under its scope. To not confuse the developers, we did not provide
+/// the default implementations here, but we are providing a macro to generate
+/// the default implementations for you.
```

Comment:
```suggestion
/// present under its scope. To avoid confusion, we do not provide the default
/// implementations here, but we are providing a macro that generates them
///  for you.
```